cmd/geth/config.go

```diff
@@ -294,7 +294,105 @@ func makeFullNode(ctx *cli.Context) (*node.Node, ethapi.Backend) {
 		// rather than working correctly until then and freezing at activation until TBC is caught up.
 		// Would rather have node delay full startup to fully sync TBC to required genesis height
 		// than startup without full information and later freeze at hVM Phase 0 activation time.
+
 		for {
+
+			// wait until we have a block header that is above hvm genesis height, then confirm we
+			// have hvm genesis block at the expected height
+			for {
+				select {
+				case <-time.After(1 * time.Second):
+					bh, _, err := vm.TBCFullNode.BlockHeaderBest(ctx.Context)
+					if err != nil {
+						log.Crit(fmt.Sprintf("error getting block header best: %s", err))
+					}
+
+					log.Trace(fmt.Sprintf("the current best block height is %d", bh))
```

Comment:
Calling `fmt.Sprintf` for a trace log is not ideal, as it will result in Sprintf being run even when this will never be logged. Instead, I believe geth uses structured logging, so something like this might be more expected (maybe look at existing code and follow the existing logging patterns):
```suggestion
					log.Trace("Best block height", "block_height", b)
```

Ideally we should follow the existing log format, and avoid unnecessary calls to `fmt.Sprintf` wherever possible, **in all other logs as well**, please.

Reply:
sure, will update

Reply:
I am only going to modify the logs that I added in this PR though, as we use this pattern throughout this repo.  I don't want this PR's scope to increase.

Reply:
hey @joshuasing I believe that I got them all, could you re-check?